### ISSUE TITLE
Bridge fails on Announcement channels

### ISSUE THREAD

**Issue by Miepee** [Half-Shot/matrix-appservice-discord#774]
Trying to setup the bridge on "Discord announcement channels" fails.
It waits a while before responding back with `There was a problem bridging that channel - has the guild owner approved the bridge?` even if it has Admin powers in the Discord Server.

**Comment by bicknrown** [comment]
Does the bot role specifically have send message access to that channel?

`channel settings -> Permissions -> bot role`
Try enabling the following if not already done;

* `Manage Webhooks`
* `Send Messages`
* `Embed Links`
* `Manage Messages`
* `Read Message History`

Then try and bridge the channel, and see if the behavior changes.

**Comment by Miepee** [comment]
> Does the bot role specifically have send message access to that channel?

Yes, it does. Not only does it have full administrator permissions, it also has every permission enabled.

**Comment by bicknrown** [comment]
Are you using the `1.0.0` release, or the `develop` branch?

If using the `1.0.0` release, try building from the `develop` branch, and see if that changes anything.

If you built from the `develop` branch, from what I could gather, the issue seems to stem from either the incorrect channel ID / guild ID being passed to the command, or the bot is for some reason unable to send messages in the requested channel.

**Comment by Miepee** [comment]
I'm actually using the hosted t2bot instance (<https://github.com/t2bot/matrix-appservice-discord>).
I asked in their matrix help room about this issue, they said that the channel type isn't handled correctly and that this is an upstream issue, so I submitted it here.

**Comment by bicknrown** [comment]
Ah, alright. Here's hoping that gets patched on the bot instance sooner than later. 
If anything, this thread helps people with other permission based issues while we wait.

**Comment by erkinalp** [comment]
Due to peculiarities of Discord's API, sending messages in announcement channels actually require `MANAGE_MESSAGES`, and those channels cannot admit incoming webhooks except for cross-posting ones.

**Comment by deepadmax** [comment]
I posted this in the incorrect place. Maybe this will add some insight.

> Currently, channels marked as Announcement Channel can't be bridged. You need to unmark it, bridge it, and mark it again. It works fine even if it's marked as such. It just can't be marked at the time of bridging. This is problematic if people are already following the channel, as it would break the follower's feed and remove the channel from their server.

I don't think it has to do with permissions; I've given the bot *all* the permissions.
The obvious way to get around the other limitation you posited is to bridge it one way, as that's what one would want anyway with such a channel. It's not necessary, though, because it works fully if set up as aforementionedly described.
